From 01c30c6743512ff285595ad67a428ed988fd9a1d Mon Sep 17 00:00:00 2001
From: Octavian Purdila <octavian.purdila@nxp.com>
Date: Mon, 12 Jun 2017 11:58:44 +0300
Subject: [PATCH 1887/5242] MLK-15047-2: ARM: imx: add IMX_SOC_IMX7

commit  35393f00fe30934a3928d0d58a43c1f66b3e404d from
https://source.codeaurora.org/external/imx/linux-imx.git

IMX_SOC_IMX7 is referenced in makefiles and kconfig but it is not
defined, so define it and select it for both IMX7D and IMX7ULP.

Fixes the following build errors:

arch/arm/mach-imx/built-in.o: In function `update_lpddr2_freq_smp':
platform-imx-dma.c:(.text+0xf7c): undefined reference to `imx_scu_base'
platform-imx-dma.c:(text+0xf88): undefined reference to `imx_scu_base'
arch/arm/mach-imx/built-in.o: In function `update_ddr_freq_imx_smp':
platform-imx-dma.c:(text+0x330c): undefined reference to `imx_scu_base'
platform-imx-dma.c:(text+0x3318): undefined reference to `imx_scu_base'
Makefile:969: recipe for target 'vmlinux' failed
make: *** [vmlinux] Error 1

Signed-off-by: Octavian Purdila <octavian.purdila@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/Kconfig |   21 ++++++++++-----------
 1 file changed, 10 insertions(+), 11 deletions(-)

diff --git a/arch/arm/mach-imx/Kconfig b/arch/arm/mach-imx/Kconfig
index 6963567..f04bc23 100644
--- a/arch/arm/mach-imx/Kconfig
+++ b/arch/arm/mach-imx/Kconfig
@@ -564,23 +564,27 @@ config SOC_IMX6ULL
 	help
 	  This enables support for Freescale i.MX6 ULL processor.
 
+config SOC_IMX7
+	bool
+	select CPU_V7
+	select ARM_GIC
+	select HAVE_IMX_MU
+	select HAVE_IMX_RPMSG
+	select RPMSG
+
 config SOC_IMX7D
 	bool "i.MX7 Dual support"
+	select SOC_IMX7
 	select PCI_DOMAINS if PCI
 	select PINCTRL_IMX7D
-	select ARM_GIC
 	select HAVE_ARM_ARCH_TIMER
 	select HAVE_IMX_ANATOP
 	select HAVE_IMX_MMDC
 	select HAVE_IMX_DDRC
 	select HAVE_IMX_SRC
 	select IMX_GPCV2
-	select HAVE_IMX_MU
-	select HAVE_IMX_RPMSG
-	select RPMSG
 	select KEYBOARD_SNVS_PWRKEY
 	select HAVE_IMX_GPCV2
-	select CPU_V7
 	select ARM_ERRATA_814220
 
 	help
@@ -611,14 +615,9 @@ config SOC_IMX6SLL
 
 config SOC_IMX7ULP
 	bool "i.MX7ULP support"
-	select CPU_V7
-	select ARM_GIC
+	select SOC_IMX7
 	select CLKSRC_IMX_TPM
-	select HAVE_IMX_MU
 	select PINCTRL_IMX7ULP
-	select HAVE_IMX_RPMSG
-	select RPMSG
-
 	help
 	  This enables support for Freescale i.MX7 Ultra Low Power processor.
 
-- 
1.7.9.5

